src/panels/config/backup/ha-config-backup.ts

```diff
@@ -0,0 +1,192 @@
+
+  @property({ attribute: false }) public route!: Route;
+
+  @state() private _backupData?: BackupData;
+
+  private _columns = memoize(
+    (narrow, _language): DataTableColumnContainer => ({
+      name: {
+        title: this.hass.localize("ui.panel.config.backup.name"),
+        sortable: true,
+        filterable: true,
+        grows: true,
+        template: (entry: string, backup: BackupContent) =>
+          html`${entry}
+            <div class="secondary">${backup.path}</div>`,
+      },
+      size: {
+        title: this.hass.localize("ui.panel.config.backup.size"),
+        width: "15%",
+        hidden: narrow,
+        filterable: true,
+        sortable: true,
+        template: (entry: number) => Math.ceil(entry * 10) / 10 + " MB",
+      },
+      date: {
+        title: this.hass.localize("ui.panel.config.backup.created"),
+        width: "15%",
+        direction: "desc",
+        hidden: narrow,
+        filterable: true,
+        sortable: true,
+        template: (entry: string) =>
+          relativeTime(new Date(entry), this.hass.locale),
+      },
+
+      actions: {
+        title: "",
+        width: "15%",
+        template: (entry: string, backup: BackupContent) =>
+          html` <ha-icon-overflow-menu
+            .hass=${this.hass}
+            .narrow=${this.narrow}
+            .items=${[
+              // Download Button
+              {
+                path: mdiDownload,
+                label: this.hass.localize(
+                  "ui.panel.config.backup.download_backup"
+                ),
+                action: () => this._downloadBackup(backup),
+              },
+              // Delete button
+              {
+                path: mdiDelete,
+                label: this.hass.localize(
+                  "ui.panel.config.backup.remove_backup"
+                ),
+                action: () => this._removeBackup(backup.slug),
+              },
+            ]}
+            style="color: var(--secondary-text-color)"
+          >
+          </ha-icon-overflow-menu>`,
+      },
+    })
+  );
+
+  private _getItems = memoize((backupItems: BackupContent[]) =>
+    backupItems.map((backup) => ({
+      name: backup.name,
+      slug: backup.slug,
+      date: backup.date,
+      size: backup.size,
+      path: backup.path,
+    }))
+  );
+
+  protected render(): TemplateResult {
+    if (!this.hass || this._backupData === undefined) {
+      return html` <hass-loading-screen></hass-loading-screen> `;
+    }
+
+    return html`
+      <hass-tabs-subpage-data-table
+        .hass=${this.hass}
+        .narrow=${this.narrow}
+        back-path="/config"
+        .route=${this.route}
+        .tabs=${configSections.backup}
+        .columns=${this._columns(this.narrow, this.hass.language)}
+        .data=${this._getItems(this._backupData.backups)}
+        .noDataText=${this.hass.localize("ui.panel.config.backup.no_bakcups")}
+      >
+        <ha-fab
```

Comment:
Should we disable the fab and change the text while a backup is being generated? Otherwise the user will not know a backup is in progress.

Reply:
No, a user should always be able to click and fail even if we know it will never work.

Reply:
Why not disable plus change text that were making a backup ?